core/templates/services/math-interactions.service.ts

```diff
@@ -183,6 +183,100 @@ export class MathInteractionsService {
     }
     return modifiedExpressionList.join('');
   }
+
+  getTerms(expressionString: string, splitByAddition = true): Array<string> {
+    let listOfTerms: Array<string> = [];
+    let currentTerm: string = '';
+    let bracketBalance: number = 0;
+    let shouldModifyTerm: boolean = false;
+    let modifyTerm = function(termString: string): string {
+      // If the shouldModifyTerm flag is set to true, we add the '-' sign,
+      // raise the term to a power of -1. This ensures that when the final
+      // list is joined by the '+'/'*' sign, it matches with the original
+      // expression. For eg.
+      // '3/10' would be split as [3, 10^(-1)] and
+      // '3-10' would be split as [3, -(10)].
+      if (splitByAddition) {
+        return '-(' + termString + ')';
+      } else {
+        return '(' + termString + ')^(-1)';
+      }
+    };
+    const delimiters: Array<string> = splitByAddition ? ['+', '-'] : ['*', '/'];
+
+    expressionString = expressionString.split(' ').join('');
+
+    // Temporarily replacing all unary negation signs with '~' so as to avoid
+    // splitting terms by them. We only need to split terms by binary
+    // subtraction signs and not unary negation signs. A '-' sign is considered
+    // to be binary subtraction iff it is preceded by an alphanumeric or a
+    // closing bracket, otherwise it is considered as unary negation operation.
+    // NOTE: The replace function is called twice to deal with cases where there
+    // might be overlapping matches.
+    // For eg. 4----5 would be converted to 4-~-~5 after the first call. So we
+    // need a second call to convert it to the desired result, which is 4-~~~5.
+    expressionString = expressionString.replace(/([^a-zA-Z0-9\)])-/g, '$1~');
+    expressionString = expressionString.replace(/([^a-zA-Z0-9\)])-/g, '$1~');
+
+    for (let i = 0; i < expressionString.length; i++) {
+      let currentVal = expressionString[i];
+      if (currentVal === '(' || currentVal === ')') {
+        bracketBalance += (currentVal === '(') ? 1 : -1;
+      }
+
+      // Split term only if we are not inside a set of parens and the current
+      // value is a delimiter.
+      if (bracketBalance === 0 && delimiters.indexOf(currentVal) !== -1) {
+        if (currentTerm.length !== 0) {
+          if (shouldModifyTerm) {
+            currentTerm = modifyTerm(currentTerm);
+            shouldModifyTerm = false;
+          }
+          listOfTerms.push(currentTerm);
+          currentTerm = '';
+        }
+        if (currentVal === delimiters[1]) {
```

Comment:
Also, isn't it better to check this initially (inside the higher if block) and finish up with `currentTerm` in this iteration itself? Since, rn you are waiting for the next iteration to do modifications, right?
You can also not use a separate `shouldModifyTerm` flag also.

Reply:
> Hmm, so you mean '-' and '/' are the terms that need modification, right? I think it'd be better if you explicitly mention that, like have 2 variables for delimiters instead of a list with appropriate names, since if at any point someone reverses that list, then this would be hard to debug.

Yes. Done.

> Also, isn't it better to check this initially (inside the higher if block) and finish up with currentTerm in this iteration itself? Since, rn you are waiting for the next iteration to do modifications, right?
You can also not use a separate shouldModifyTerm flag also.

Actually, the `shouldModifyTerm` is supposed to indicate if we should be modifying the next term. For instance, say the expression is `3/10`, then when the `/` sign is encountered, the `shouldModifyTerm` flag is set to true, thereby indicating that the next term we get is supposed to be modified (add - sign or raise to a power of -1). This is why we need a flag and can't just modify the current term because we won't know if we need to modify the current term without using a flag.
To avoid this confusion, I've renamed the flag from `shouldModifyTerm` to `shouldModifyNextTerm`.